ghcide/session-loader/Development/IDE/Session.hs

```diff
@@ -527,7 +526,10 @@ cradleToOptsAndLibDir cradle file = do
 emptyHscEnv :: IORef NameCache -> FilePath -> IO HscEnv
 emptyHscEnv nc libDir = do
     env <- runGhc (Just libDir) getSession
+#if !MIN_VERSION_ghc(9,0,0)
+    -- This causes ghc9 to crash
```

Comment:
Are you calling `initLoaderState? Is there a GHC bug report?

Reply:
No, there is no GHC bug report (that I know of). I don't really know what's wrong. But it seems to work when I don't call it.

Reply:
Can you explain what that function does and why we are calling it?

If I remove the conditional so it is called on ghc9 as well, I get this error:
```
 ghcide: Couldn't find a target code interpreter. Try with -fexternal-interpreter
 ```
 
 It seems like the `hsc_env` variable was not set in the `HscEnv` we get from `getSession`. I don't really know why or really what any of this means.
 
 I'm guessing that this issue was introduced by this commit: https://gitlab.haskell.org/ghc/ghc/commit/18757cab04c5c5c48eaceea19469d4811c5d0371
 
 For some reason, `hsc_interp` is only set when you call `setSessionDynFlags`, despite that seeming like a relatively unrelated function.

Reply:
Maybe I should do this instead, so I can call `initDynLinker`?
```
    env <- runGhc (Just libDir) $ getSessionDynFlags >>= setSessionDynFlags >> getSession
```
The tests seem to succeed both with that change as well as when not calling `initDynLinker` at all.

Reply:
That was apparently premature, some tests fail (e.g.  the `ghcide.cradle.multi` tests) with the
```
env <- runGhc (Just libDir) $ getSessionDynFlags >>= setSessionDynFlags >> getSession
```
version, but all tests succeed when I simply don't run `initLoaderState` at all.

What is the harm of not running it/what is it needed for?

Reply:
Ping @wz1000